### PR TITLE
Merge dev into master

### PR DESCRIPTION
**Pull request by alimaktabi**

### CODE REVIEW

File: app/contribution-hub/ConstraintDetailsModal.tsx [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+  return (
+    <div className="flex flex-col gap-2 mt-5 relative">
+      <div
+        className="absolute -top-14 cursor-pointer z-[999]"
+        onClick={handleBackToConstraintListModal}
+      >
+        <Icon
+          iconSrc="/assets/images/provider-dashboard/arrow-left.svg"
+          className="cursor-pointer z-[999999]"
+        />
```
Comment (alimaktabi) [inline app/contribution-hub/ConstraintDetailsModal.tsx:164]: @abbasnosrati 

this icon does not have `position: relative` or `position: absolute` applied. remove `z-[999999]`

File: app/contribution-hub/ConstraintDetailsModal.tsx [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+      <CreateParams
+        constraint={constraint}
+        setRequirementParamsList={setRequirementParamsList}
+        requirementParamsList={requirementParamsList}
+        constraintFile={constraintFile}
+        setConstraintFile={setConstraintFile}
+        allChainList={allChainList}
+      />
+      <div className="mb-4">{constraint.description}</div>
+      <div className="text-error text-[10px] min-h-[15px]">{errorMessage}</div>
```
Comment (alimaktabi) [inline app/contribution-hub/ConstraintDetailsModal.tsx:199]: use `text-2xs` instead

File: app/contribution-hub/ConstraintDetailsModal.tsx [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+      <div
+        onClick={handleAddRequirement}
+        className="flex cursor-pointer  bg-gray40 text-[14px] font-semibold text-white h-[44px] border-2 border-gray70 rounded-xl items-center justify-center mb-2"
```
Comment (alimaktabi) [inline app/contribution-hub/ConstraintDetailsModal.tsx:202]: use `text-sm` instead

File: app/contribution-hub/ConstraintDetailsModal.tsx [hunk 4]
```diff
@@ -0,0 +1,273 @@
...
+}
+
+const ConstraintDetailsModal = ({
+  constraint,
+  handleBackToConstraintListModal,
+  requirementList,
+  insertRequirement,
+  updateRequirement,
+  allChainList,
+}: DetailsModal) => {
```
Comment (alimaktabi) [inline app/contribution-hub/ConstraintDetailsModal.tsx:39]: replace the typing with `FC<DetailsModal>` instead

File: app/contribution-hub/ConstraintDetailsModal.tsx [hunk 5]
```diff
@@ -0,0 +1,273 @@
...
+
+  const handleChangeCollection = (address: string) => {
+    setReqNftAddress(address);
+    setRequirementParamsList({
+      ...requirementParamsList,
+      ["COLLECTION_ADDRESS"]: address,
+    });
+  };
+
+  if (constraint.params.length === 0) return;
```
Comment (alimaktabi) [inline app/contribution-hub/ConstraintDetailsModal.tsx:233]: do not return `undefined`. return `null` instead

File: app/contribution-hub/SelectMethodInput.tsx [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+
+interface Prop {
+  setRequirementParamsList: (e: any) => void;
+  requirementParamsList: any;
+}
+
+const SelectMethodInput = ({
+  setRequirementParamsList,
+  requirementParamsList,
+}: Prop) => {
```
Comment (alimaktabi) [inline app/contribution-hub/SelectMethodInput.tsx:18]: use `FC` type instead

File: app/contribution-hub/SelectMethodInput.tsx [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+
+  const handleSelectMethod = () => {
+    setSelectedMethod("Minimum Amount");
+    setShowItems(false);
+  };
+
+  interface MethodProp {
+    minimum: number;
+    maximum: number;
+  }
```
Comment (alimaktabi) [inline app/contribution-hub/SelectMethodInput.tsx:30]: move outside of component decleration

File: app/contribution-hub/gas-tap/components/Content.tsx [hunk 8]
```diff
@@ -0,0 +1,158 @@
...
+import Link from "next/link";
+import RoutePath from "@/utils/routes";
+
+const GasTapContent = () => {
+  return (
+    <div>
+      <div>
+        <div className="flex flex-col md:flex-row  items-center justify-between ">
+          <SearchInput className="w-full md:w-1/3 st" />
+          <div className="provider-dashboard__status justify-center mt-5 md:mt-0 flex h-[40px] text-[12px] items-center align-center text-gray90 bg-gray40 border-2 border-gray30 rounded-xl w-full  md:w-auto">
```
Comment (alimaktabi) [inline app/contribution-hub/gas-tap/components/Content.tsx:18]: replace `text-[12px]`

File: app/contribution-hub/gas-tap/components/Content.tsx [hunk 9]
```diff
@@ -0,0 +1,158 @@
...
+            <div>verified</div>
+            <div>rejected</div>
+            <div>finished</div>
+          </div>
+        </div>
+        <div className="refill-token h-auto md:h-[78px] mt-4 flex w-full justify-between overflow-hidden items-center">
+          <div className="flex flex-col sm:flex-row justify-between w-full items-center py-5 px-7 text-white">
+            <div className="flex items-center relative">
+              <div>
+                <p className="text-[16px] font-semibold">
```
Comment (alimaktabi) [inline app/contribution-hub/gas-tap/components/Content.tsx:30]: replace `text-[16px]`

File: app/contribution-hub/CsvFileInput.tsx [hunk 10]
```diff
@@ -0,0 +1,99 @@
...
+const CsvFileInput = ({
+  setRequirementParamsList,
+  requirementParamsList,
+  setConstraintFile,
+  constraintFile,
+}: Prop) => {
+  const [isUploadedFileValid, setIsUploadedFileValid] =
+    useState<boolean>(false);
+
+  console.log(constraintFile);
```
Comment (alimaktabi) [inline app/contribution-hub/CsvFileInput.tsx:20]: remove console log

File: app/contribution-hub/gas-tap/create/page.tsx [hunk 11]
```diff
@@ -0,0 +1,13 @@
+"use client";
```
Comment (alimaktabi) [inline app/contribution-hub/gas-tap/create/page.tsx:1]: remove this line

File: app/contribution-hub/ConstraintDetailsModal.tsx (hunk 1 above)
Comment (abbasnosrati) [reply app/contribution-hub/ConstraintDetailsModal.tsx:164]: Done!

File: app/contribution-hub/ConstraintDetailsModal.tsx (hunk 3 above)
Comment (abbasnosrati) [reply app/contribution-hub/ConstraintDetailsModal.tsx:202]: fixed

File: app/contribution-hub/SelectMethodInput.tsx (hunk 7 above)
Comment (abbasnosrati) [reply app/contribution-hub/SelectMethodInput.tsx:30]: done!

File: app/contribution-hub/gas-tap/create/page.tsx (hunk 11 above)
Comment (abbasnosrati) [reply app/contribution-hub/gas-tap/create/page.tsx:1]: done

File: app/contribution-hub/CsvFileInput.tsx (hunk 10 above)
Comment (abbasnosrati) [reply app/contribution-hub/CsvFileInput.tsx:20]: done

File: app/contribution-hub/ConstraintDetailsModal.tsx (hunk 5 above)
Comment (abbasnosrati) [reply app/contribution-hub/ConstraintDetailsModal.tsx:233]: fixed

**Comment by abbasnosrati** [comment]
Done!

**Pull request merged**